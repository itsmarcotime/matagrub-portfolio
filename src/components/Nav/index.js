import React from "react";

function Nav() {

    const categories = [
        {name: "work", description: "links to other project I have worked on"},
        {name: "resume", description: "link to my resume in case you are curious"},
    ];

    function categorySelected(name) {
        console.log(`${name} clicked`)
    };

    return (
        <header>
            {/* here needs animated pictures(think of credits type of animation) to make page look busy could fit 
            into an h2 and placed about the nav */}
            <nav>
                <ul className="flex-row">

                    <li className="mx-2">
                        <a data-testid="home" href="/">
                            Home
                        </a>
                    </li>

                    <li className="mx-2">
                        <a data-testid="about" href="#about">
                            About
                        </a>
                    </li>

                    {categories.map((category) => (
                        <li
                          className="mx-1"
                          key={category.name}
                        >
                            <span onClick={() => categorySelected(category.name)}>
                                {category.name}
                            </span>
                        </li>  
                    ))}

                    <li>
                        <span>Contact</span>
                    </li>
                </ul>
            </nav>

            {/* here needs icon or logo image */}
        </header>
    )
}

export default Nav;