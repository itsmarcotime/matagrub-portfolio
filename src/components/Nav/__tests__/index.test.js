import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Nav from '..';

afterEach(cleanup);

describe('Nav component', () => {
    // baseline render
    it('renders', () => {
        render(<Nav />);
    });

    //snapshot test
    it('matches snapshot', () => {
        const { asFragment } = render(<Nav />);

        expect(asFragment()).toMatchSnapshot();
    });
});

//add more tests when adding animated image to nav, 20.2.5

describe('links are shown', () => {
    it('inserts text into the links', () => {
        const { getByTestId } = render(<Nav />);

        expect(getByTestId('home')).toHaveTextContent('Home');
        expect(getByTestId('about')).toHaveTextContent('About');
    });
});