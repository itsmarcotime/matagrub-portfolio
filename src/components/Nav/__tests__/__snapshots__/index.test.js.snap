// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component matches snapshot 1`] = `
<DocumentFragment>
  <header>
    <nav>
      <ul
        class="flex-row"
      >
        <li
          class="mx-2"
        >
          <a
            data-testid="home"
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="mx-2"
        >
          <a
            data-testid="about"
            href="#about"
          >
            About
          </a>
        </li>
        <li
          class="mx-1"
        >
          <span>
            work
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            resume
          </span>
        </li>
        <li>
          <span>
            Contact
          </span>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
